### PR TITLE
Fix the Travis MacOS builds (update XCode to version 9.4 and use matplotlib 'Agg' backend)

### PR DESCRIPTION
**Pull request by GenevieveBuckley**
Running builds with Xcode 6.4 in Travis CI is deprecated and was removed in January 2019. 

If we remove the `osx_image: ` line from `.travis.yml` then our builds will use the default XCode version (currently that's XCode 9.4). See here for more details: https://docs.travis-ci.com/user/reference/osx/#xcode-version

### CODE REVIEW

**Comment by jakirkham** [comment]
Sorry for the troubles with CI, @GenevieveBuckley. Thanks for stepping in and trying to address them.

The Travis CI image was chosen mostly to align with conda-forge at one point in time, but it is not strictly necessary here. Happy to change it in whatever way works best.

It seems like newer versions of Conda may have broken some things. Feel free to pin the Miniconda download URL to some 4.5 version. Will try to raise this upstream when I get a chance.

**Comment by GenevieveBuckley** [comment]
So I think the problem might lie in `.travis_support/create_env.sh`. It seems that variables `BASH_SOURCE`, and therefore `SCRIPT_DIR` are blank, which means that when we `conda env create -f path/to/environment.yml` we're not actually using the absolute path to the environment yml file.

Any suggestions? I can't replicate this behaviour locally, so I've been mucking around with travis https://travis-ci.com/GenevieveBuckley/travis-test/

Side note: I have also noticed some weirdness when using `conda env create` with the `--path` flag. This I could replicate locally. Using both `--path` and `--file` together results in a conda environment being created, but no name being attached to it (despite the name being specified in the environment yml file).

**Comment by GenevieveBuckley** [comment]
Yep, for some reason we're not getting the result we expect from $BASH_SOURCE. It *should* be `.travis_support`, so when I specify that explicitly conda is at least able to find the environment yml files.

One problem down, more to go.

There's a problem with the version of `atomicwrites`, a dependency of one of our dependencies (maybe pytest?). Conda wants to install atomicwrites version 1.3.0, but only 1.2.1 appears to be available for Mac: https://anaconda.org/conda-forge/atomicwrites/files

```
Executing transaction: / failed
ERROR conda.core.link:_execute(507): An error occurred while installing package 'conda-forge::atomicwrites-1.3.0-py_0'.
FileNotFoundError(2, "No such file or directory: '/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/tmpehbs1ln3/tst_py37/bin/python3.7'")
Attempting to roll back.
Rolling back transaction: - done
FileNotFoundError(2, "No such file or directory: '/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/tmpehbs1ln3/tst_py37/bin/python3.7'")
```

**Comment by GenevieveBuckley** [comment]
Update:
* Fiddling with travis in another repo, I could build the environments if I installed `dask`, `scikit-image`, `pytest` and `pytest-flake8` with pip instead of conda. This worked for all python versions (2.7, 3.5, 3.6, and 3.7) using the same pinned dependency version numbers used here. It's frustrating that it doesn't seem to behave similarly here.
* The dpl conda environment fails in conjunction with python 2.7 (because miniconda has trouble installing conda forge backports). I'm not entirely sure what this environment is used for & why it's needed before coveralls.
* It doesn't seem to matter much whether we use miniconda 4.5.4 or 4.5.12
* I can't reproduce this error locally at all on a Mac (with conda, not miniconda)

**Comment by GenevieveBuckley** [comment]
I think I need to give up on this for a while, fyi

**Comment by jakirkham** [comment]
Thanks for pushing it this far. I'll try to find some time to take a look. Unfortunately it might be a little bit before that happens.

**Comment by GenevieveBuckley** [comment]
I think I have it working now - travis builds are all green here: https://travis-ci.com/GenevieveBuckley/dask-image/builds/110839172. I'll update this PR to reflect that in a bit, and have added my notes below.

### 1. backports in the python 2.7 env
This is fixed if `defaults` is added to the list of conda channels in the yml file

### 2. Setting the matplotlib backend explicitly to fix problem in python 2.7 env
Pytest test collection fails because (test_imread imports pims, which imports matplotlib, which has a fit about python not being a framework build but only for the python 2.7 environment). This is fixed by importing matplotlib and setting the backend explicitly before any import of pims:

```python
import matplotlib
matplotlib.use('Agg')  # noqa: E402
import pims
```
It seems Agg is the most common & appropriate matplotlib backend. Another plausible option here is TkAgg.  Happy for people to weigh in with opinions here.

I sincerely hope this does not cause problems with the CircleCI or Appveyor builds. We'll see!

### 3. dpl.yml conda environment
The dpl.yml conda environment can't find file conda-forge::docopt-0.6.2-py_1
This is fixed by adding `defaults` to the list of conda channels in the yml file.

**Comment by GenevieveBuckley** [comment]
Why do we have a special environment just for coveralls, instead of just calling coveralls straight away?

**Comment by GenevieveBuckley** [comment]
Ok, this is ready for review @jakirkham 

1. Currently I've pinned XCode to the default version 9.4 on Travis. We could also bump it to version 10.2 if you feel that would slightly future proof the tests. [I've tested the builds pass here with xcode 10.2](https://travis-ci.com/GenevieveBuckley/dask-image/builds/110979281).

2. Coveralls decreases slightly due to the two added lines: `matplotlib.use('Agg')`. Is this acceptable, or can you think of a good way to address this? (We could possibly delay importing matplotlib/pims?)

File: .travis_support/environments/dpl.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 name: dask_image_dpl_env
 
 channels:
+  - defaults
```
Comment (jakirkham) [inline .travis_support/environments/dpl.yml]: Is this still needed?

Comment (GenevieveBuckley) [reply .travis_support/environments/dpl.yml]: As it turns out, yes.

I watched the builds pass, then tried removing the defaults channel and saw it fail:
* https://travis-ci.org/GenevieveBuckley/dask-image/jobs/558679819
* https://travis-ci.org/GenevieveBuckley/dask-image/jobs/558679820

This was the relevant part:
```
+++/Users/travis/miniconda/bin/conda env create -p /var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/tmpmiq6qeat/dpl -f .travis_support/environments/dpl.yml
Solving environment: done
==> WARNING: A newer version of conda exists. <==
  current version: 4.5.12
  latest version: 4.7.5
Please update conda by running
    $ conda update -n base -c defaults conda
Downloading and Extracting Packages
cffi-1.12.3          | 210 KB    | : 100% 1.0/1 [00:00<00:00,  6.08it/s]               
certifi-2019.6.16    | 147 KB    | : 100% 1.0/1 [00:00<00:00, 15.88it/s]
pyopenssl-19.0.0     | 81 KB     | : 100% 1.0/1 [00:00<00:00, 14.39it/s]
urllib3-1.25.3       | 187 KB    | : 100% 1.0/1 [00:00<00:00,  6.27it/s]
docopt-0.6.2         | 14 KB     | : 100% 1.0/1 [00:00<00:00, 18.66it/s]
python-3.6.7         | 20.6 MB   | : 100% 1.0/1 [00:04<00:00,  4.80s/it]               
six-1.12.0           | 22 KB     | : 100% 1.0/1 [00:00<00:00, 18.22it/s]
pycparser-2.19       | 173 KB    | : 100% 1.0/1 [00:00<00:00,  7.13it/s]
chardet-3.0.4        | 190 KB    | : 100% 1.0/1 [00:00<00:00,  7.62it/s]               
idna-2.8             | 132 KB    | : 100% 1.0/1 [00:00<00:00, 13.31it/s]
coverage-4.5.1       | 211 KB    | : 100% 1.0/1 [00:00<00:00,  7.21it/s]
pip-18.0             | 1.8 MB    | : 100% 1.0/1 [00:00<00:00,  1.28it/s]               
pysocks-1.7.0        | 26 KB     | : 100% 1.0/1 [00:00<00:00, 16.63it/s]
asn1crypto-0.24.0    | 154 KB    | : 100% 1.0/1 [00:00<00:00,  8.48it/s]
yaml-0.1.7           | 79 KB     | : 100% 1.0/1 [00:00<00:00, 18.79it/s]
cryptography-2.7     | 566 KB    | : 100% 1.0/1 [00:00<00:00,  2.88it/s]               
setuptools-41.0.1    | 620 KB    | : 100% 1.0/1 [00:00<00:00,  3.67it/s]               
wheel-0.31.1         | 62 KB     | : 100% 1.0/1 [00:00<00:00, 15.64it/s]
coveralls-1.3.0      | 12 KB     | : 100% 1.0/1 [00:00<00:00, 27.28it/s]
requests-2.22.0      | 84 KB     | : 100% 1.0/1 [00:00<00:00, 13.92it/s]
pyyaml-5.1.1         | 166 KB    | : 100% 1.0/1 [00:00<00:00,  9.40it/s]
Preparing transaction: done
Verifying transaction: done
Executing transaction: failed
ERROR conda.core.link:_execute(507): An error occurred while installing package 'conda-forge::docopt-0.6.2-py_1'.
FileNotFoundError(2, "No such file or directory: '/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/tmpmiq6qeat/dpl/bin/python3.6'")
Attempting to roll back.
Rolling back transaction: done
FileNotFoundError(2, "No such file or directory: '/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/tmpmiq6qeat/dpl/bin/python3.6'")
```

**Comment by GenevieveBuckley** [comment]
🎉

**Comment by GenevieveBuckley** [comment]
FYI: I'll merge this in by July 30th. @jakirkham if you feel there was anything we didn't cover at the sprints let me know, otherwise I'll go ahead with that then.

**Comment by jakirkham** [comment]
Thanks @GenevieveBuckley!

Sorry for being slow here. Was in the process of moving near the end of July. That's (mostly) done now so should be able to catch up here soon.

**Comment by GenevieveBuckley** [comment]
> Sorry for being slow here. 

No need to apologize, you're pretty busy! The PyConAU sprints are on right now which is the reason for the burst of activity here.

**Comment by jakirkham** [comment]
Ah great! Do people seem interested over there? Did you happen to give a talk?

**Comment by GenevieveBuckley** [comment]
I gave a 5 minute lightning talk on a completely unrelated topic. I've been having conversations with (mostly geo or satellite imagery) people about their use cases. It's almost midnight here, so I'm going to go to sleep, but I'll send an email later on this week.

**Comment by jakirkham** [comment]
Sounds like fun. Yeah would be interesting to hear more about what they are looking for.

No worries. Sleep is good. 🙂

**Pull request merged**